ee-feature-pack/common/src/main/resources/modules/system/layers/base/com/fasterxml/jackson/core/jackson-databind/main/module.xml

```diff
@@ -21,12 +21,16 @@
   ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
   ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
   -->
-<module xmlns="urn:jboss:module:1.5" name="com.fasterxml.jackson.core.jackson-databind">
+<module xmlns="urn:jboss:module:1.9" name="com.fasterxml.jackson.core.jackson-databind">
+
      <resources>
         <artifact name="${com.fasterxml.jackson.core:jackson-databind}"/>
     </resources>
-
     <dependencies>
+        <module name="java.desktop"/>
+        <module name="java.logging"/>
+        <module name="java.sql"/>
+        <module name="java.xml"/>
```

Comment:
Everywhere you identified proper Java's module dependencies like in this case @boris-unckel 
you can eliminate dependency on deprecated 'wrapper' javax.api module that
was bringing in all dependencies you identified here.

Reply:
@ropalka I have removed the dependencies, there is a comment in place now: `<!--WFLY-14219 Remove deprecated <module name="javax.api"/> -->`
The JDK8 fails _are not caused by the removal_. I have run a test, undoing the removal everywhere, JDK8 still fails within the same tests and reason. One can reproduce that JDK11 runs without error and JDK8 fails. Can you please give advice:
1. Howto make jboss modules give a relevant and good trace to analyse the issue?
2. What rules are relevant for the module descriptors to work with both JDK8 and JDK11? 
- Is the order of modules relevant: **all** java.* and/or jdk.* modules first? all java.* and/or jdk.* modules last? 
- Does the jdk.unsupported module work with Java 8 or is the WildFly-Core module "sun.jdk" still needed (even it is deprecated?)
- Other rules?

Sample for the first (others completely different)
`2020-12-23 19:22:01,592 WARN  [org.jboss.modules.define] (Weld Thread Pool -- 3) Failed to define class org.wildfly.test.integration.microprofile.health.MicroProfileHealthCheckOperationTestCase in Module "deployment.MicroProfileHealthTestCase.war" from Service Module Loader: java.lang.NoClassDefFoundError: Failed to link org/wildfly/test/integration/microprofile/health/MicroProfileHealthCheckOperationTestCase (Module "deployment.MicroProfileHealthTestCase.war" from Service Module Loader): org/wildfly/test/integration/microprofile/health/MicroProfileHealthTestBase
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:832)
	at org.jboss.modules.ModuleClassLoader.doDefineOrLoadClass(ModuleClassLoader.java:424)
	at org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:555)
	at org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:339)
	at org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:126)
	at org.jboss.modules.Module.loadModuleClass(Module.java:753)
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:247)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at org.jboss.as.weld.WeldModuleResourceLoader.classForName(WeldModuleResourceLoader.java:68)
	at org.jboss.weld.bootstrap.AnnotatedTypeLoader.loadClass(AnnotatedTypeLoader.java:82)
	at org.jboss.weld.bootstrap.AnnotatedTypeLoader.loadAnnotatedType(AnnotatedTypeLoader.java:62)
	at org.jboss.weld.bootstrap.FastAnnotatedTypeLoader.loadAnnotatedType(FastAnnotatedTypeLoader.java:108)
	at org.jboss.weld.bootstrap.BeanDeployer.addClass(BeanDeployer.java:87)
	at org.jboss.weld.bootstrap.ConcurrentBeanDeployer$1.doWork(ConcurrentBeanDeployer.java:55)
	at org.jboss.weld.bootstrap.ConcurrentBeanDeployer$1.doWork(ConcurrentBeanDeployer.java:52)
	at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:62)
	at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:55)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:513)`

Reply:
@ropalka I have tested some more: For every module which is imported with service somewhere I added a provides-service-withclass section in the module descriptor, based on the services in the corresponding JAR. This does not help for any of the JDK8 cases, but totally breaks the JDK11 build. I assume modules which import several other modules with their services re-export them, currently implicit and with my change lacking the explicit way.
I will stop for now and wait for your analysis and advice.